wetterdienst/util/network.py

```diff
@@ -45,9 +98,63 @@ def list_remote_files(url: str, recursive: bool) -> List[str]:
             folders.append(urljoin(url, f))
 
     if recursive:
-        files_in_folders = [list_remote_files(folder, recursive) for folder in folders]
+        files_in_folders = [
+            list_remote_files_legacy(folder, recursive) for folder in folders
+        ]
 
         for files_in_folder in files_in_folders:
             files.extend(files_in_folder)
 
     return files
+
+
+# v2: "Remote directory index" implementation based on FSSPEC.
+def list_remote_files_fsspec(
+    url: str, recursive: bool = False, ttl: CacheExpiry = CacheExpiry.FILEINDEX
+) -> List[str]:
+    """
+    A function used to create a listing of all files of a given path on the server.
+
+    The default ttl with ``CacheExpiry.FILEINDEX`` is "5 minutes".
+
+    Args:
+        :param url:         The URL which should be searched for files.
+        :param recursive:   Definition if the function should iteratively list files
+                            from sub folders.
+        :param ttl:         The cache expiration time.
+
+    Returns:
+        A list of strings representing the files from the path.
+    """
+
+    # Acquire filesystem instance.
+    filesystem = NetworkFilesystemManager.get(ttl=ttl)
+
+    # Recursively list remote directory.
+    if not recursive:
+        remote_urls = filesystem.find(url)
+    else:
+        remote_urls = filesystem.expand_path(url, recursive=recursive)
```

Comment:
I think .find() is totally sufficient and will already scan files in subdirectories. This way recursive can be omitted totally from this function.

Reply:
Ah. Maybe that's the reason I used `.glob()` before because I _intentionally_ wanted to prevent using recursive operations in all occasions in order to reduce overhead?

Reply:
So, let's keep the implementation like it is now?

Reply:
I've just added 467d0db1b. Thanks for this suggestion.

Reply:
Knowing that you have some additional arguments how to improve this spot, I will still be merging it and look forward to any improvements from your pen through a subsequent patch.